test/unit-tests.test.ts

```diff
@@ -899,6 +899,21 @@ test('Infinity', async () => {
   `)).resolves.toBe(-Infinity);
 });
 
+test('LHS w/ another assign op', async () => {
+  await expect(run(`
+    const a = [1, 3, 5, 7];
+    let i = 0;
+    a[++i] += 3;
+    i;
+  `)).resolves.toBe(1);
+  await expect(run(`
+    const a = [1, 3, 5, 7];
+    let i = 0;
+    a[++i] = a[++i] + 3;
```

Comment:
Jeez, what a crazy example. Good job!
Suggest you also check the "correct" value of the array - although I am clueless as to what that should be.